File: request_test.py
import os
import anthropic
from dotenv import load_dotenv

load_dotenv()

client = anthropic.Anthropic(api_key=os.getenv("ANTHROPIC_API_KEY"))


def parse_sentences(text: str) -> list[str]:
    sentences = []
    for line in text.splitlines():
        line = line.strip()
        if not line:
            continue
        if line[0].isdigit():
            parts = line.split(".", 1)
            if len(parts) > 1:
                line = parts[1].strip()
        sentences.append(line)
    return sentences


def get_shortest_sentence(sentences: list[str]) -> str:
    return min(sentences, key=len)


def get_similar_sentences(input_sentence: str) -> list[str]:
    prompt = f"""Generate 10 semantically similar sentences to the following sentence.
Make them as short as possible.

Sentence: "{input_sentence}"

Return ONLY a numbered list of 10 sentences."""

    message = client.messages.create(
        model="claude-haiku-4-5-20251001",
        max_tokens=1024,
        messages=[{"role": "user", "content": prompt}],
    )

    return parse_sentences(message.content[0].text)


def run_test(user_input: str) -> str:
    import time
    start = time.time()

    sentences = get_similar_sentences(user_input)
    output = ""

    if len(sentences) < 10:
        output += "Warning: fewer than 10 sentences returned\n"

    shortest = get_shortest_sentence(sentences)

    output += "\nGenerated sentences:\n"
    for s in sentences:
        output += f"- {s}\n"

    output += "\nShortest sentence:\n"
    output += shortest + "\n"
    output += f"took {time.time() - start:.3f}s\n"

    return output


if __name__ == "__main__":
    user_input = input("Enter a sentence: ")
    print(run_test(user_input))

File: src/request_test.js
const API_URL = "https://router.huggingface.co/v1/chat/completions";
const API_KEY = process.env.VITE_HF_API_KEY || "";

export function parseSentences(text) {
    const lines = text.split("\n");
    const sentences = [];

    for (let line of lines) {
        line = line.trim();
        if (!line) continue;

        if (/^\d/.test(line)) {
            const parts = line.split(".");
            if (parts.length > 1) {
                line = parts.slice(1).join(".").trim();
            }
        }
        sentences.push(line);
    }
    return sentences;
}

export function getShortestSentence(sentences) {
    return sentences.reduce((a, b) => a.length <= b.length ? a : b);
}

export async function getSimilarSentences(inputSentence) {
    const prompt = `
    Generate 10 semantically similar sentences to the following sentence.
    make them as short as possible.

    Sentence: "${inputSentence}"

    Return ONLY a numbered list of 10 sentences.
    `;

    const headers = {
        "Authorization": `Bearer ${API_KEY}`,
        "Content-Type": "application/json"
    };

    const payload = {
        messages: [
            {
                role: "user",
                content: prompt
            }
        ],
        model: "meta-llama/Llama-3.1-8B-Instruct:novita"
    };

    const response = await fetch(API_URL, {
        method: "POST",
        headers: headers,
        body: JSON.stringify(payload)
    });

    if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();
    const text = data.choices[0].message.content;

    return parseSentences(text);
}

export async function runTest(userInput, outputElement) {
    const start = performance.now();
    try {
        const sentences = await getSimilarSentences(userInput);
        
        let output = "";
        if (sentences.length < 10) {
            output += "Warning: fewer than 10 sentences returned\n";
        }

        const shortest = getShortestSentence(sentences);

        output += "\nGenerated sentences:\n";
        for (const s of sentences) {
            output += `- ${s}\n`;
        }

        output += "\nShortest sentence:\n";
        output += shortest + "\n";
        const end = performance.now();
        output += `took ${(end - start) / 1000}s\n`;
        
        if (outputElement) {
            outputElement.textContent = output;
        } else {
            console.log(output);
        }

    } catch (e) {
        console.error("Error:", e);
        if (outputElement) {
            outputElement.textContent = `Error: ${e.message}`;
        }
    }
}
